hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -816,6 +867,18 @@ public void startCommitWithTime(String commitTime) {
         String commitActionType = table.getCommitActionType();
         activeTimeline.createInflight(
             new HoodieInstant(true, commitActionType, commitTime));
+
+      // create temporary folder if needed
```

Comment:
This is a one time thing.. so suggest, we just make the .hoodie/tmp folder be initialized with the other metadata and remove it from here,, (same principle.. this file should be oblivious to whats going on underneath for COW)